integration/index_test.go

```diff
@@ -0,0 +1,178 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/awslabs/soci-snapshotter/soci"
+	"github.com/awslabs/soci-snapshotter/util/dockershell"
+	"github.com/containerd/containerd/platforms"
+)
+
+type testImageIndex struct {
+	imgName         string
+	platform        string
+	imgRef          string
+	sociIndexDigest string
+}
+
+func prepareSociIndices(t *testing.T, sh *dockershell.Shell) []testImageIndex {
+	testImages := []testImageIndex{
+		{
+			imgName:  "ubuntu:latest",
+			platform: "linux/arm64",
+		},
+		{
+			imgName:  "alpine:latest",
+			platform: "linux/amd64",
+		},
+		{
+			imgName:  "nginx:latest",
+			platform: "linux/arm64",
+		},
+		{
+			imgName:  "drupal:latest",
+			platform: "linux/amd64",
+		},
+	}
+
+	for i, img := range testImages {
+		platform := platforms.DefaultSpec()
+		if img.platform != "" {
+			var err error
+			platform, err = platforms.Parse(img.platform)
+			if err != nil {
+				t.Fatalf("could not parse platform: %v", err)
+			}
+		}
+		imgInfo := dockerhub(img.imgName, withPlatform(platform))
+		img.imgRef = imgInfo.ref
+		img.sociIndexDigest = optimizeImage(sh, imgInfo)
+		testImages[i] = img
+	}
+
+	return testImages
+}
+
+func TestSociIndexInfo(t *testing.T) {
+	t.Parallel()
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+	rebootContainerd(t, sh, "", "")
+
+	testImages := prepareSociIndices(t, sh)
+
+	var sociIndex soci.Index
+
+	for _, img := range testImages {
+		t.Run(img.imgName, func(t *testing.T) {
+			rawJSON := sh.O("soci", "index", "info", img.sociIndexDigest)
+			if err := json.Unmarshal(rawJSON, &sociIndex); err != nil {
+				t.Fatalf("invalid soci index from digest %s: %v", img.sociIndexDigest, rawJSON)
+			}
+			if sociIndex.MediaType != soci.OCIArtifactManifestMediaType {
+				t.Fatalf("invalid soci index media type: expected: %s, actual: %s", soci.OCIArtifactManifestMediaType, sociIndex.MediaType)
+			}
+			if sociIndex.ArtifactType != soci.SociIndexArtifactType {
+				t.Fatalf("invalid soci index artifact type: expected: %s, actual: %s", soci.SociIndexArtifactType, sociIndex.ArtifactType)
+			}
```

Comment:
I think we probably want to test for `subject`, `annotations` and `blobs` fields.

Reply:
made some refactor and reuse some code from `create` integration test. It checks all now.